pl_examples/basic_examples/sync_bn.py

```diff
@@ -0,0 +1,196 @@
+"""
+Sync-bn with DDP (GPU)
+
+This code is to verify that batch statistics are synchronized across GPUs using sync-bn.
+When sync_bn_backend is set to 'torch' or 'apex', the training loop should run for 3 iterations.
+When sync_bn_backend is set to None, the code should result in an AssertionError.
+"""
+import os
+import math
+from argparse import ArgumentParser
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import pytorch_lightning as pl
+
+import torchvision.transforms as transforms
+from torchvision.datasets import MNIST
+from torch.utils.data import random_split
+from torch.utils.data import DataLoader
+
+
+pl.seed_everything(234)
+EPSILON = 1e-12
```

Comment:
consider using https://github.com/PyTorchLightning/pytorch-lightning/blob/a55c481d5d27028bd34024705d79cbea66c48d32/pytorch_lightning/utilities/__init__.py#L19

Reply:
@Borda this comes out to be quite a large value '0.000977'. numpy.finfo(numpy.float32).eps is 1.1920929e-07.

Reply:
yes, but 1e-12 fro AMP with float16 is 0
cc: @williamFalcon